x-pack/plugins/spaces/public/spaces_context/context.tsx

```diff
@@ -7,29 +7,29 @@
 
 import * as React from 'react';
 
+import type { CoreStart } from 'src/core/public';
+
 import type { SpacesManager } from '../spaces_manager';
-import type { ShareToSpacesData } from '../types';
-import type { KibanaServices, SpacesReactContext, SpacesReactContextValue } from './types';
+import type { SpacesData } from '../types';
+import type { SpacesReactContext, SpacesReactContextValue } from './types';
 
 const { useContext, createElement, createContext } = React;
 
-const context = createContext<Partial<SpacesReactContextValue<KibanaServices>>>({});
+const context = createContext<Partial<SpacesReactContextValue<Partial<CoreStart>>>>({});
```

Comment:
(not related to this PR, was already there: Why was the spaces context extending `KibanaServices`, and why do we need to have it now extends `Partial<CoreStart>`? Why is `SpacesReactContextValue` extending anything?

Reply:
I based this implementation off of the Kibana React Context:

https://github.com/elastic/kibana/blob/4584a8b570402aa07832cf3e5b520e5d2cfa7166/src/plugins/kibana_react/public/context/context.tsx#L22

https://github.com/elastic/kibana/blob/4584a8b570402aa07832cf3e5b520e5d2cfa7166/src/plugins/kibana_react/public/context/types.ts#L14

The `useSpaces` context also provides access to core start services so you don't need to use two different context providers in the client consumer. I guess it was written with this generic type so you can specify what core services you need to use.